builder/builder-next/exporter/export.go

```diff
@@ -73,8 +72,21 @@ func (e *imageExporterInstance) Name() string {
 	return "exporting to image"
 }
 
-func (e *imageExporterInstance) Export(ctx context.Context, ref cache.ImmutableRef, opt map[string][]byte) (map[string]string, error) {
-	if config, ok := opt[exporterImageConfig]; ok {
+func (e *imageExporterInstance) Export(ctx context.Context, inp exporter.Source) (map[string]string, error) {
+
+	if len(inp.Refs) > 1 {
+		return nil, fmt.Errorf("exporting multiple references to image store is currently unsupported")
+	}
+	ref := inp.Ref
+	if ref != nil && len(inp.Refs) == 1 {
+		return nil, fmt.Errorf("invalid exporter input: Ref and Refs are mutually exclusive")
+	}
+	// only one loop
+	for _, v := range inp.Refs {
+		ref = v
+	}
+
+	if config, ok := inp.Metadata[exporterImageConfig]; ok {
```

Comment:
If you are handling the `Refs > 0` case then the config is using a different path. Check the buildkit image exporter for the value.

Reply:
@tonistiigi fixed. PTAL